vcf-dumper/vcf-dumper-lib/src/main/java/embl/ebi/variation/eva/vcfdump/BiodataVariantToVariantContextConverter.java

```diff
@@ -38,100 +35,105 @@
         }
-    }
 
-    private String getContextNucleotideFromCellbaseCachingRegions(String chromosome, int contextNucleotidePosition, Region region) throws CellbaseSequenceDownloadError {
-        int regionStart = region.getStart() - 1;
-        if (regionSequence == null) {
-            // if an indel start is the first nucleotide of the region, we will need the previous nucleotide, so we are adding
-            // the preceding nucleotide to the region (region.getStart()-1)
-            int regionEnd = region.getEnd();
-            try {
-                regionSequence = cellbaseClient.getSequence(new Region(chromosome, regionStart, regionEnd));
-            } catch (Exception e) {
-                throw new CellbaseSequenceDownloadError("Error getting from Cellbase sequence for Region " + chromosome + ":"
-                        + regionStart + "-" + regionEnd, e);
-            }
+        String[] srcLineFields = srcLine.get().split("\t", 5);
+
+        // get the relative position of the context nucleotide in the source line REF string
+        int positionInSrcLine = Integer.parseInt(srcLineFields[1]);
+        // the context nucleotide is generally the one preceding the variant
+        boolean prependContextNucleotideToVariant = true;
+        int relativePositionOfContextNucleotide = variant.getStart() - 1 - positionInSrcLine;
+        // if there is no preceding nucleotide in the source line, the context nucleotide will be "after" the variant
+        if (relativePositionOfContextNucleotide < 0) {
+            relativePositionOfContextNucleotide = variant.getStart() + variant.getReference().length() - positionInSrcLine;
+            prependContextNucleotideToVariant = false;
         }
-        String nucleotide = getNucleotideFromRegionSequence(contextNucleotidePosition, regionStart, regionSequence);
-        return nucleotide;
+
+        // get context nucleotide and add it to the variant
+        String contextNucleotide = getContextNucleotideFromSourceLine(srcLineFields, relativePositionOfContextNucleotide);
+        variant = addContextNucleotideToVariant(variant, contextNucleotide, prependContextNucleotideToVariant);
+
+        return variant;
+    }
+
+
+    private String getContextNucleotideFromSourceLine(String[] srcLineFields, int relativePositionOfContextNucleotide) {
+        String referenceInSrcLine = srcLineFields[3];
+        return referenceInSrcLine.substring(relativePositionOfContextNucleotide, relativePositionOfContextNucleotide + 1);
     }
 
-    private String getNucleotideFromRegionSequence(int nucleotidePosition, int regionStart, String regionSequence) {
-        int relativePosition = nucleotidePosition - regionStart;
-        return regionSequence.substring(relativePosition, relativePosition + 1);
+    private Variant addContextNucleotideToVariant(Variant variant, String contextNucleotide, boolean prependContextNucleotideToVariant) {
```

Comment:
Would be good to link some references on why the coordinates are modified this way